### PR TITLE
Second version of Clusterly front-end

### PR DESCRIPTION
**Pull request by chenraff**
- Add inputs and connect the front-end to the server. 
- Add an option to delete queries by dragging the corresponding bubble to a delete button.
- Add an option to add new cluster by clicking on the plus button and choosing the new cluster name.
- Add category input component and update server classes to support it.

### CODE REVIEW

**Review by mayascha — APPROVED** [review]
Looks like a very exiting changes, hopefully you'll be able to deploy soon so I can see how it works

File: trendly/src/app/terms-chips-input/terms-chips-input.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    if (index >= 0) {
+      this.terms.splice(index, 1);
+      this.emitTerms();
+    }
+  }
+
+  /**
+   * Emits the terms to the parent 'inputs' component when the input value
+   * changes.
+   */
```
Comment (ewebdev) [inline trendly/src/app/terms-chips-input/terms-chips-input.component.ts:54]: redundant, in class methods/properties only document details that are unclear.

File: trendly/src/app/terms-chips-input/terms-chips-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      this.emitTerms();
+    }
+    // Reset the input value
+    if (input) {
+      input.value = '';
+    }
+  }
+
+  /**
+   * Removes the given term from the terms list
```
Comment (ewebdev) [inline trendly/src/app/terms-chips-input/terms-chips-input.component.ts:41]:
```suggestion
   * Removes the given term from the terms list.
```

File: trendly/src/app/clusters-section/clusters-section.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+
+  /**
+   * Checks addScales adds successfully all the scales defined in Scales enum.
+   */
+  it('should add all needed scales', () => {
+    (component as any).addScales();
+    for (const scale in Scales) {
+      if (!isNaN(+scale)) {
+        expect(typeof (component as any).scales.get(+scale)).toBe('function');
```
[the remark this answers is not in the document]
Comment (chenraff) [reply trendly/src/app/clusters-section/clusters-section.component.spec.ts:102]: Thanks for the resources! I'll read more about it.

**Review by ewebdev — APPROVED** [review]

**Pull request merged**